python/pyarrow/_s3fs.pyx

```diff
@@ -91,85 +97,40 @@ cdef class S3Options:
                 '`access_key` is not set.'
             )
         elif access_key is not None or secret_key is not None:
-            self.options = CS3Options.FromAccessKey(
+            options = CS3Options.FromAccessKey(
                 tobytes(access_key),
                 tobytes(secret_key)
             )
         else:
-            self.options = CS3Options.Defaults()
+            options = CS3Options.Defaults()
 
         if region is not None:
-            self.region = region
+            options.region = tobytes(region)
         if scheme is not None:
-            self.scheme = scheme
+            options.scheme = tobytes(scheme)
         if endpoint_override is not None:
-            self.endpoint_override = endpoint_override
+            options.endpoint_override = tobytes(endpoint_override)
         if background_writes is not None:
-            self.background_writes = background_writes
-
-    cdef inline CS3Options unwrap(self) nogil:
-        return self.options
-
-    @property
-    def region(self):
```

Comment:
Not a blocker, but it would be nice if you could still query those options from Python.